File: 1678. Goal Parser Interpretation/Solution.py
class Solution:
    def interpret_MK1(self, command: str) -> str:
        return command.replace('()', 'o').replace('(al)', 'al')

    def interpret_MK2(self, command: str) -> str:
        res = []
        for i, c in enumerate(command):
            if c == 'G':
                res.append('G')
            elif c == ')':
                if command[i - 1] == '(':
                    res.append('o')
                else:
                    res.append('al')
        return ''.join(res)

File: 1678. Goal Parser Interpretation/Solution.cpp
#include <string>
using namespace std;

class Solution
{
public:
    string interpret(string command)
    {
        string res;
        for (size_t i = 0; i < command.size(); i++) {
            if (command[i] == 'G')
                res.push_back(command[i]);
            else if (command[i] == ')') {
                if (command[i - 1] == '(')
                    res.push_back('o');
                else
                    res += "al";
            }
        }
        return res;
    }
};
